lib/dialects/abstract/parser-store.js

```diff
@@ -0,0 +1,14 @@
+class ParserStore extends Map {
```

Comment:
Do we really want to subclass map?

Reply:
wouldn't we? It really is a map, and was before it just proxied very weirdly to add this one extra function 🤔 also, it's really easy to type it well